hbase-client/src/main/java/org/apache/hadoop/hbase/zookeeper/ZooKeeperHelper.java

```diff
@@ -61,6 +61,7 @@ public static ZooKeeper ensureConnectedZooKeeper(ZooKeeper zookeeper, int timeou
     while (!zookeeper.getState().isConnected()) {
       Threads.sleep(1);
       if (stopWatch.elapsed(TimeUnit.MILLISECONDS) > timeout) {
+        zookeeper.close();
```

Comment:
I suspect you added this to cover the case when constructor of `HACK_UNTIL_ZOOKEEPER_1897_ZooKeeperMain` failes and close() not getting called. I think you might want to do this within the constructor by putting `ZooKeeperHelper.ensureConnectedZooKeeper(this.zk, 30000);` in a try-catch block to keep responsibility inside the class.

Reply:
Thank you for reviewing
I have changed the code to handle ZooKeeperConnectionException in the constructor.